### PR TITLE
pbench agent scripts: fix using of -p with scp command

### LINKED ISSUE #591: scp commands will fail when -p <port> is used as ssh_opts in the pbench-agent-config
ssh uses -p flag to connect to a remote host while scp uses -P, this will lead to failure. For example, when the ssh_opts is set to use -p 2022 in the pbench-agent.cfg file, the scp command in the pbench-move-results script which copies the results on to the remote server fails since it uses -P flag for port not -p.

### PR DESCRIPTION
**Pull request by tenstormavi**
fixes #591

### CODE REVIEW

**Comment by tenstormavi** [comment]
[scp commands will fail when -p <port> is used as ssh_opts in the pbench-agent-config](https://trello.com/c/9tyI2hSy/33-scp-commands-will-fail-when-p-port-is-used-as-sshopts-in-the-pbench-agent-config)

File: agent/bench-scripts/pbench-dbench [hunk 1]
```diff
@@ -347,7 +347,7 @@ for thread_count in `echo $thread_counts | sed -e s/,/" "/g`; do
 						if [ "$client" != "127.0.0.1" ]; then
 							ssh $ssh_opts $client mkdir -p $benchmark_results_dir
 							ssh $ssh_opts $client mkdir -p $dir/$client_nr
-							scp $ssh_opts $benchmark_client_cmd_file $client:$benchmark_client_cmd_file >/dev/null
+							$scp_opts $ssh_opts $benchmark_client_cmd_file $client:$benchmark_client_cmd_file >/dev/null
```
Comment (ndokos) [inline agent/bench-scripts/pbench-dbench]: scp $scp_opts ....

and get rid of ssh_opts.

File: agent/bench-scripts/pbench-netperf [hunk 2]
```diff
@@ -461,7 +461,7 @@ for protocol in `echo $protocols | sed -e s/,/" "/g`; do
 										client=`echo $clients | cut -d, -f$i`
 										xml_file="$benchmark_run_dir/$iteration/netperf-config-$server.xml"
 										ssh $ssh_opts $client mkdir -p $benchmark_results_dir
-										scp $ssh_opts $xml_file $client:$xml_file
+										$scp_opts $ssh_opts $xml_file $client:$xml_file
```
Comment (ndokos) [inline agent/bench-scripts/pbench-netperf]: scp $scp_opts $xml_file ....

File: agent/bench-scripts/pbench-migrate [hunk 3]
```diff
@@ -255,7 +255,7 @@ function do_migrate() {
 		echo creating VM on destination host using libvirt
 		ssh $dest_host virsh undefine $vm
 		ssh $src_host virsh dumpxml $vm >/tmp/$vm.xml
-		scp /tmp/$vm.xml $dest_host:/tmp/$vm.xml
+		scp $scp_opts /tmp/$vm.xml $dest_host:/tmp/$vm.xml
```
Comment (ndokos) [inline agent/bench-scripts/pbench-migrate]: The ssh calls should probably have $ssh_opts appended.

**Comment by ndokos** [comment]
> Also, there should be another commit for the unit test files, no?

Eh, maybe not: the output has probably not changed on the existing ones, but there should be a couple of new ones with -p/-P enabled.

**Comment by tenstormavi** [comment]
@ndokos updated the PR. PTAL

**Pull request merged**